### PR TITLE
[Misc] Add `get_name` method to attention backends

### PR DESCRIPTION
**Pull request by WoosukKwon**
This PR adds the name strs to the attention backends so that we can use the names to identify them (without importing the actual class).

### CODE REVIEW

File: vllm/attention/backends/flashinfer.py [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-try:
-    import flashinfer
-    from flash_attn import flash_attn_varlen_func
-    from flashinfer import BatchDecodeWithPagedKVCacheWrapper
-except ImportError:
-    flashinfer = None
-    flash_attn_varlen_func = None
-    BatchDecodeWithPagedKVCacheWrapper = None
-
+import flashinfer
```
Comment (comaniac) [inline vllm/attention/backends/flashinfer.py:4]: Does this import make `flashinfer` as the must-to-have dependency?

Comment (WoosukKwon) [reply vllm/attention/backends/flashinfer.py:4]: Not really. The attention backends are lazily imported by `get_attn_backend`. Currently, the flashinfer backend is not imported unless the user specifies it via an env variable.

Comment (comaniac) [reply vllm/attention/backends/flashinfer.py:4]: I see. Thanks for the clarification :D

**Review by comaniac — APPROVED** [review]

**Review by LiuXiaoxuanPKU — APPROVED** [review]
LGTM. Thanks for fixing the flashinfer import!

**Pull request merged**